src/main/java/app/servlets/AddRecordServlet.java

```diff
@@ -0,0 +1,190 @@
+                        if(name.equals("author"))
+                        {
+                            author=value;
+                        }
+                        if(name.equals("genre"))
+                        {
+                            genre=value;
+                        }
+                        if(name.equals("publisher"))
+                        {
+                            publisher=value;
+                        }
+                        if(name.equals("date"))
+                        {
+                            date=value;
+                        }
+                        if(name.equals("description"))
+                        {
+                            description=value;
+                        }
+                        if(name.equals("pages"))
+                        {
+                            pages=Integer.parseInt(value);
+                        }
+                        if(name.equals("isbn"))
+                        {
+                            isbn=Long.parseLong(value);
+                        }
+                        if(name.equals("total_amount"))
+                        {
+                            total_amount=Integer.parseInt(value);
+                        }
+                    }
+                    else
+                    {
+                        int id = 1;
+                        filename = new File(item.getName()).getName();
+                        int i = filename.lastIndexOf('.');
+                        if (i != -1) {
+                            String filename2 = filename.substring(0, i) + id;
+                            id++;
+                            String ext = filename.substring(i, filename.length());
+                            filename = filename2 + ext;
+                        }
+                        try {
+                            file = new File(UPLOAD_DIRECTORY + File.separator + filename);
+                            item.write(file);
+                            byte[] bArray = readFileToByteArray(file);
+                            String base64Image = Base64.encodeBase64String(bArray);
+                            book.setCover64(base64Image);
+                        }catch (Exception e){
+                            e.printStackTrace();
+                        }
+                    }
+                }
+            }
+
+            book.setTitle(title);
+            book.setAuthor(author);
+            book.setGenre(genre);
+            book.setPublisher(publisher);
+            book.setDate(date);
+            book.setPages(pages);
+            book.setIsbn(isbn);
+            book.setDescription(description);
+            book.setAmount(total_amount);
+            book.setStatus(st_1);
+        }
+
+        insertBook(book);
+
+        req.setAttribute("book", book);
+        req.getRequestDispatcher("views/full_book_page.jsp").forward(req, resp);
+    }
+
+    private void insertBook(Book book) {
```

Comment:
in here, if insertion resulted in an error, you still redirect user to a book page, even though nothing was saved. This will only result in more exceptions thrown and won't tell user what went wrong